### DISCUSSION TITLE
Why "component" instead "group"?

### DISCUSSION THREAD

**Discussion by jmeistrich** [nativewind/nativewind#48] [Q&A] [answered]
I haven't dug into it yet, but just curious why deviate from tailwind's naming. Does it work differently than "group"?

**Comment by marklawlor** [comment, marked as answer]
**Group**

```
<div className="group">
  <div className="group">
    <p className="group-hover:text-red-500">hello, world!</p>
  </div>
</div>
```

In this case, the `<p>` will go red when either `div` is hovered. 

**Component**

```
<Pressable className="component">
  <Pressable className="component">
    <Text className="component-hover:text-red-500">hello, world!</Text>
  </Pressable>
</Pressable>
```

In this case, the `<Text>` will go red when only when its parent "component" is hovered.

**Reasoning**

RN doesn't allow for all interaction listeners on `<Text />` and other components, forcing you to wrap them in `<Pressables />`. But you want those interactions scoped to only that `<Pressable />`

Imagine you have card  (that uses `<Pressable />`)

```
<Card className="component hover:bg-key-500">
  <Text className="component-hover:color-white">This is my card</Text>
  <Pressable className="component">
    <Text className="text-sky-500 component-hover:color-white">Ok</Text>
  </Pressable>
</Card>
```

In this example, you only want the "Ok" button to change color when the user hovers over the button. But if we used group, it would change when they hovered over the card.

I'm looking to add another class soon that will allow components to "inherit" their parents interactions and still exploring ways to port this to CSS StyleSheets.

Also `group` is coming - its basically `component` but nested groups are ignored (or they inherit by default)

**Comment by marklawlor** [comment]
Also documentation on this is coming soon that will go into a little bit more detail.

**Comment by jmeistrich** [comment]
Thanks! That makes sense. `component` sounds like it will be very useful :)

**Comment by danstepanov** [comment]
Closing this discussion, if this is still relevant, feel free to open a new one.
